python/python_direct/ops.cpp

```diff
@@ -2874,6 +2874,78 @@ tuple[TensorView, TensorView, TensorView, TensorView]
     A tuple of (output, log_sumexp, philox_seed, philox_offset).
       )",
       py::return_value_policy::reference);
+  ops.def(
+      "sdpfa_bwd",
+      [](TensorView* grad_output,
+         TensorView* query,
+         TensorView* key,
+         TensorView* value,
+         TensorView* output,
+         TensorView* log_sumexp,
+         Val* dropout_p,
+         Val* is_causal,
+         TensorView* philox_seed,
+         TensorView* philox_offset,
+         Val* scale) -> decltype(auto) {
```

Comment:
A nitpick not related to this PR. I like the plain pointer type signature.

In our sequence of PRs for direct binding. I think I'm seeing inconsistency across using `nullptr` vs using `std::nullopt` and matching signature as `std::optional<T*>`.

I don't have a strong preference on the style. But would be nice to converge on one. :wink:

Reply:
I think only SCAN uses `std::optional<T*>`. All the other ops should use the plain pointer with `nullptr`.
https://github.com/NVIDIA/Fuser/blob/ddf9f9cbee005db23f9931d506c9a16bd8080f3e/python/python_direct/ops.cpp#L245

Reply:
I pushed [85ef3fb](https://github.com/NVIDIA/Fuser/pull/4913/commits/85ef3fb37c7598565057250b2a107d3ac2389be0) to change existing `std::optional<TensorView*>` to `TensorView*` in this PR.